server/src/inmantals/server.py

```diff
@@ -241,8 +241,27 @@ def create_tmp_project(self) -> str:
         with open(os.path.join(inmanta_project_dir, "project.yml"), "w+") as fd:
             yaml.dump(metadata, fd)
 
+        def _get_name_spaces(curdir: str, prefix: str) -> List[str]:
+            """
+            Returns a list of all inmanta namespaces living under a root directory
+            """
+            files: List[str] = []
+            init_cf = os.path.join(curdir, "_init.cf")
+            if not os.path.exists(init_cf):
+                return files
+
+            files.append(prefix)
+            for entry in os.listdir(curdir):
+                sub_path = os.path.join(curdir, entry)
+                if os.path.isdir(sub_path):
+                    files.extend(_get_name_spaces(sub_path, prefix + "::" + entry))
+
+            return files
+
+        name_spaces = _get_name_spaces(os.path.join(self.folder_path, "model"), module_name)
         with open(os.path.join(inmanta_project_dir, "main.cf"), "w+") as fd:
-            fd.write(f"import {module_name}\n")
+            for name in name_spaces:
+                fd.write(f"import {name}\n")
```

Comment:
Why did we need this?

Reply:
We need to import all sub-modules, otherwise they won't be know to the project when performing the anchormap computation, (and so navigation wouldn't work for them). I realized this when testing on the big junos-qfx module, navigation was only working on the .cf file living at the root of the folder